[DRAFT][FIX] account, web: onboarding

igbe-odoo opened the pull request:
Change few ui for mobile purpose :

1 - hide drag & drop button on mobile view and replace it by upload button 2 - Fix an issue with your_pointer, to let him pop on top of nav bar instead of behind 3 - Add a nev div on nav bar list item, to point items more precisely in tours.

task-4497104

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/196797.png)](https://mergebot.odoo.com/odoo/odoo/pull/196797)

rfr-odoo commented on addons/account/static/src/components/bill_guide/bill_guide.scss:
```diff
@@ -23,3 +23,15 @@
     border-color: $o-brand-primary;
     background-color: #F2EDF0;
 }
+
+@media only screen and (max-width:768px) {
+    .mobile_hide {
+        display: none !important;
+    }
+}
+
+@media only screen and (min-width:768px) {
+    .desktop_hide {
+        display: none !important;
+    }
+}
```
use Bootstrap rules instead creating same CSS rules again :wink: 
c.f.: https://getbootstrap.com/docs/5.3/utilities/display/#notation
e.g.:
```html
<span class="btn account_drag_drop_btn pe-none d-none d-md-inline">Drag &amp; drop</span>
```

rfr-odoo commented on addons/web/static/src/webclient/navbar/navbar.xml:
```diff
@@ -68,7 +68,9 @@
             'py-2': isNested,
         }">
-        <t t-esc="section.name"/>
+        <div style="width: fit-content;">
+            <t t-esc="section.name"/>
+        </div>
```
Why not use a span ?
```suggestion
            <span t-esc="section.name"/>
```

pparidans commented on addons/web_tour/static/src/tour_pointer/tour_pointer.scss:
```diff
@@ -48,7 +48,7 @@
     --TourPointer__translate-y: 0;
 
-    z-index: $zindex-tooltip;
+    z-index: $zindex-tooltip + 30;
```
Why `+ 30` ? :shrug:

igbe-odoo replied:
Forgot to remove it, mb, I already change the z-index in `tour_pointer.scss` (I do this because the pointer is rendering behind the navbar in mobile)

pparidans replied:
Ok then. But why this specific value ?

igbe-odoo replied:
I choose +20 (not +30, it was a mistake) to override the already existing `z-index: $zindex-tooltip + 10;` in the sidebar style.

I didn't test other tours in the ui but you can find the issue on my runbot by running the `account_accountant_tour` onboarding tour in mobile view, you will see that without this z-index, the pointer is showing behide the sidebar (you need to delete all the bills of your company to trigger the steps)

pparidans replied:
In summary, the `z-index` related to your issue are the following :
- sidebar uses `z-index: $zindex-tooltip + 10;`
- overlay-item uses `z-index: $zindex-modal;`
- tour pointer uses `z-index: $zindex-tooltip;`

Then what's the purpose of raising the overlay-item's z-index *when it contains a tour pointer* and not the tour pointer itself ?

On the other side, I'm wondering why we use/if we still need to use this specific value for the sidebar instead of `$zindex-modal`... @stefanorigano or @rfr-odoo Any thoughts about that ?

igbe-odoo replied:
I just found it cleaner to raise the whole overlay item in front of the sidebar instead of only the pointer itself. But I'll change that.
And yes, using `zindex-modal` for the sidebar would fix this `z-index` problem, as the tooltip will always be on top of the sidebar. :thinking:
I'll follow your call.

stefanorigano replied:
Hello @pparidans   👋

> On the other side, I'm wondering why we use/if we still need to use this specific value for the sidebar instead of $zindex-modal... @stefanorigano or @rfr-odoo Any thoughts about that ?

About this specific point (I'm not checking the PR) I think that we should probably review our z-index map taking in account how components behave. Even when we don't use bootstrap classes directly, it would make sense to ensure that our components adhere with BS z-index map, for consistency and because that's a good example to follow actually ([ref.](https://github.com/twbs/bootstrap/blob/main/scss/_variables.scss#L1133)).

```scss
$zindex-dropdown:                   1000 !default;
$zindex-sticky:                     1020 !default;
$zindex-fixed:                      1030 !default;
$zindex-offcanvas-backdrop:         1040 !default;
$zindex-offcanvas:                  1045 !default;
$zindex-modal-backdrop:             1050 !default;
$zindex-modal:                      1055 !default;
$zindex-popover:                    1070 !default;
$zindex-tooltip:                    1080 !default;
$zindex-toast:                      1090 !default;
```

In the case of a sidebar ( + BurgerMenu +  BottomSheet (one day)), it makes sense to use `$zindex-offcanvas` since it's the bootstrap component that behaves more similarly and modals should be rendered "on top" of that 👍

svfu-odoo reviewed (approved):
LGTM from the accounting side.
@h4818 for final review

pparidans reviewed (changes requested):
Hi :wave: ,

First, as you have 3 (related but) separated fixes, I would assume to have dedicated commits for each of them (ease the readability of your changes, their documentation...).

Next, about each one :

> 1 - hide drag & drop button on mobile view and replace it by upload button

I would use a `t-if="..."` instead of responsive classes to actually remove the unneeded elements from the DOM.
Also, does it need to be done *only* for small screens ? Or more generally "mobile-like" ones (smartphone, tablets...) ? I guess the issue is more related to the "drag & drop" part than the screen size...

> 2 - Fix an issue with your_pointer, to let him pop on top of nav bar instead of behind

Is it the "navbar" or the "sidebar" ? :thinking:
Also, as discussed before, I see your point and the head hatch related to the `z-index` value management between components... In this case (and as we are in a stable version), I would suggest to match as much as possible the existing values (i.e. the one used by the sidebar) and document it in your commit so we can refer to it while cleaning up the `z-index` values used later on, in master...
In this context, your selector made sense, even if I'm wondering about potential regression in some edge case... but, well, couldn't demonstrate one, so let's try it.
Also, I think that "just" matching the `z-index` value of the sidebar should be enough (as in a stacking context, not only the `z-index` value has importance, but the order of the elements in the DOM too). Have you tried it ?

> 3 - Add a nev span on nav bar list item, to point items more precisely in tours.

Not sure to understand your point here (at least not without more information) : what is the current behaviour ? How is it not precise "enough" ? And what does this additional `<span>` provides ?

igbe-odoo commented:
Hello @pparidans :wave:,

Thanks for your feedback, I'll create 3 commits as you asked.

Regarding your 3 comments:

> 2 - Fix an issue with your_pointer, to let him pop on top of nav bar instead of behind

It's the sidebar, sorry for the confusing.
I've already tried with the same z-index on both sidebar and pointer (which is equal to 1090, computed as `zindex-tooltip + 10` in `.o_burger_menu, .o_app_menu_sidebar` classes), and the pointer is still behind. The only way I found to have the pointer on top of the sidebar is by putting `zindex-tooltip + 20` on the pointer.

> 3 - Add a nev span on nav bar list item, to point items more precisely in tours.

Actually, the pointer is pointing at the center of the html element, and the only html element we have in sidebar items is a `li`, with a width equal to the sidebar width. But as we want to point the text and not the center of the sidebar, I've added this `span` to wrap the text, so I can point directly the `span` in my tour.

I've updated the commit messages as well to be cleaner, I would appreciate to have your feedback on this :slightly_smiling_face:

igbe-odoo commented:
> when a tour pointer should be hidden by the sidebar (because it targets an element in the background, like the "New" button), this commit raises it above everything, thus pointing to another element than the one it was meant to.

I didn't think about it... That's kinda crappy indeed. Shouldn't we dynamically give the pointer the same z-index as the element it's pointing to? I don't know if it's possible though, just wondering.

> I see your point. But do we really need to do it in stable (as it actually change the DOM) ? It sounds more like an [IMP] to me... (and straight <span> tags without any classes or anything always made me wonder if it is the right choice for the job...)

Maybe we should keep it as it was in stable and do this change in master if it's more stable friendly. Or maybe is it possible to fix this issue with only scss rules without adding a `span` ? Didn't try it much yet, but maybe there's something to do in that direction. What's your opinion on this? Any ideas? :thinking:

igbe-odoo commented:
Hello @pparidans, any news about the z-index stuff? :thinking:
